### PR TITLE
[My Habits] User can't 'Mark as done' inprogress habits that were acquired ones

### LINKED ISSUE #5558: [My Habits] User can't 'Mark as done' inprogress habits that were acquired ones
**Environment:** Windows 10 Pro, Google Chrome 111.0.5563.112
**Reproducible:** always
**Build found:** 29.03.2023

**Preconditions**
User is authorized
User have at least one acquired habit and the same one inprogress

**Steps to reproduce**
1. Go to https://greencity.testgreencity.ga/swagger-ui.html#
2. Find **habit-assign-controller**
3. Controller POST/habit/assign/{habitId}/enroll/{date}
image

**Actual result**
Button "Mark as done" don't work
500 error on Swagger
image

**Expected result**
User can 'Mark as done' habit
Controller POST/habit/assign/**{habitId}**/enroll/{date} work by **habitAssignId**

### PR DESCRIPTION
**Pull request by sergoliarnik**
## Summary Of Issue :
Refactor POST/habit/assign/{habitId}/enroll/{date}, so it accepts habitAssignId instead habitId

**Issue Link**
https://github.com/ita-social-projects/GreenCity/issues/5558

## Summary Of Changes :
1) refactor enrollHabit controller method and service method
2) refactored old tests and add new ones

## CHECK LIST
- [x] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: core/src/main/java/greencity/controller/HabitAssignController.java [hunk 1]
```diff
@@ -359,27 +359,27 @@ public ResponseEntity<HabitAssignManagementDto> updateAssignByHabitId(
...
+     * @param locale        - needed language code.
      * @return {@link HabitStatusCalendarDto}.
      */
-    @ApiOperation(value = "Enroll by habit id that is assigned for current user.")
+    @ApiOperation(value = "Enroll habit assign by habitAssignId for current user.")
     @ApiResponses(value = {
         @ApiResponse(code = 200, message = HttpStatuses.OK, response = HabitAssignDto.class),
         @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
         @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
         @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND)
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/HabitAssignController.java:377]: What response will we have if HabitAssign with `habitAssignId` does not belong to the current user?

File: service-api/src/main/java/greencity/service/HabitAssignService.java [hunk 2]
```diff
@@ -197,15 +197,15 @@ List<UserShoppingAndCustomShoppingListsDto> getListOfUserAndCustomShoppingListsW
...
-     * Method to enroll {@code Habit}.
+     * Method to enroll {@code HabitAssign} by habitAssignId.
      *
-     * @param habitId  {@code Habit} id to enroll.
-     * @param userId   {@code User} id.
-     * @param dateTime {@link LocalDate} dateTime we want enroll.
-     * @param language {@link String} of language code value.
+     * @param habitAssignId {@code HabitAssign} id to enroll.
+     * @param userId        {@code User} id.
+     * @param dateTime      {@link LocalDate} dateTime we want enroll.
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/service/HabitAssignService.java]: `LocalDate` does not store or represent a time or time-zone https://docs.oracle.com/javase/8/docs/api/java/time/LocalDate.html

File: service/src/test/java/greencity/service/HabitAssignServiceImplTest.java [hunk 3]
```diff
@@ -291,14 +296,6 @@ void updateStatusByHabitIdAndUserId() {
             habitAssignService.updateStatusByHabitIdAndUserId(1L, 1L, habitAssignStatDto));
     }
 
-    @Test
-    void enrollHabitThrowWrongIdException() {
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/HabitAssignServiceImplTest.java:295]: why did you remove this test?

File: dao/src/main/java/greencity/repository/HabitAssignRepo.java [hunk 4]
```diff
@@ -266,12 +266,11 @@ List<HabitAssign> findAllHabitAssignsByStatusAndHabitId(@Param("status") HabitAs
...
-     * @param userId        {@link Long} id.
      * @return {@link HabitAssign} instance.
      */
     @Query(value = "FROM HabitAssign ha"
         + " JOIN FETCH ha.habit h JOIN FETCH h.habitTranslations ht"
         + " JOIN FETCH ht.language l"
-        + " WHERE ha.id = :habitAssignId AND ha.user.id = :userId")
-    Optional<HabitAssign> findByHabitAssignIdAndUserId(Long habitAssignId, Long userId);
+        + " WHERE ha.id = :habitAssignId")
+    Optional<HabitAssign> findByHabitAssignId(Long habitAssignId);
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/repository/HabitAssignRepo.java]: Why can't we use the [findById]( https://github.com/ita-social-projects/GreenCity/blob/dev/dao/src/main/java/greencity/repository/HabitAssignRepo.java#L31) method?

**Review by RomaMocherniuk — APPROVED** [review]

**Review by juseti — APPROVED** [review]

**Review by LiliaMokhnatska — APPROVED** [review]

**Review by ABbondar — APPROVED** [review]

**Pull request merged**